portal_pages/templatetags/portal_extras.py

```diff
@@ -11,6 +11,66 @@ def make_unique(value, unique_var): # Only one argument.
     else:
         return value
 
+@register.filter
+def remove_from_string(value, value_to_remove): 
+    """Remove a value from a list of values, ie. in order to unselect a Country from a filter of Countries"""
+    if value.find("," + value_to_remove) >= 0:
+        value = value.replace("," + value_to_remove, "")
+    elif value.find(value_to_remove) >= 0:
+        value = value.replace(value_to_remove, "")
+
+    return value
+
+@register.inclusion_tag('portal_pages/tags/filter_list.html', takes_context=True)
+def display_filter_list(context, filter_type, tags_header, items, request_list):
+
+    context.request.tags_header = tags_header # ie. countries
+    context.request.tags_header_title = tags_header.title() # ie. Countries
+    
```

Comment:
I'm not sure what issue you were seeing with accessing stuff in the context, the normal pattern for accessing request in context here would be to do something like "request = context['request']" and then proceed to use the request local variable in the function, which would remove a lot of repetitive context. prefixes on request below...